elliptic-curve/src/scalar/nonzero.rs

```diff
@@ -199,6 +199,36 @@ where
     }
 }
 
+impl<C> Mul<NonZeroScalar<C>> for NonZeroScalar<C>
+where
+    C: Curve + ScalarArithmetic,
+{
+    type Output = Self;
+
+    fn mul(self, other: Self) -> Self {
+        // Assumes that the multiplication is modulo a prime,
+        // so the product of two non-zero scalars is also non-zero.
+        let scalar = self.scalar * other.scalar;
+        debug_assert!(!bool::from(scalar.is_zero()));
+        NonZeroScalar { scalar }
+    }
```

Comment:
It should be possible to implement one of these in terms of the other, and tag it `#[inline]` so the code isn't repeated between them.